packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -92,3 +95,40 @@ export function renderMeshOwner(resourceAccount: string, meshAccount: string) :
     ? meshAccount
     : undefined;
 }
+
+/**
+ * This is the helper method to check if properties for passed match are undefined.
+ * This returns true if all properties for an object are undefined.
+ */
+export function areMatchPropertiesUndefined(match: HttpRouteMatch | GrpcRouteMatch | GrpcGatewayRouteMatch | undefined): boolean {
+  let isEmpty = true;
+  for (const property in match) {
+    if (property && (property !== 'rewriteRequestHostname' && property !== 'pathRewrite')) {
```

Comment:
Hmm. Shouldn't you also be checking if the value of the property is not `undefined`?

Also, this condition on `'rewriteRequestHostname'` and `'pathRewrite'` makes this method very complicated... The name doesn't really fit anymore as written.

Reply:
Changed the `if` statement to check `match.hasOwnProperty(property)`